### ISSUE TITLE
Provide implementation for storage space verifier (needs more info)

### ISSUE THREAD

**Issue by ataulm** [novoda/download-manager#223]
[Empty implementation is being used](https://github.com/novoda/download-manager/blob/v2/library/src/main/java/com/novoda/downloadmanager/FileDownloader.java#L75)

I thought this is writing a block of something to the disk to see we actually have enough space before we bother downloading, but it seems to "verify" the space just before it writes a chunk. @ouchadam can you give more info and edit this issue pls?

**Comment by ouchadam** [comment]
[original impl](https://github.com/novoda/download-manager/blob/8133b2a3bf0b1c07065ff8cccaa2c97cbdae7fbb/library/src/main/java/com/novoda/downloadmanager/lib/StorageSpaceVerifier.java) which delegates to a modfied AOSP [storage manager](https://github.com/novoda/download-manager/blob/8133b2a3bf0b1c07065ff8cccaa2c97cbdae7fbb/library/src/main/java/com/novoda/downloadmanager/lib/StorageManager.java) which looks like it's reading the directories and checking against available space thresholds

```java
class StorageSpaceVerifier implements SpaceVerifier {

    private final StorageManager storageManager;
    private final int destination;
    private final String fileName;

    public StorageSpaceVerifier(StorageManager storageManager, int destination, String fileName) {
        this.storageManager = storageManager;
        this.destination = destination;
        this.fileName = fileName;
    }

    @Override
    public void verifySpacePreemptively(int count) throws StopRequestException {
        storageManager.verifySpaceBeforeWritingToFile(destination, fileName, count);
    }

    @Override
    public void verifySpace(int count) throws StopRequestException {
        storageManager.verifySpace(destination, fileName, count);
    }
}
```

**Comment by ataulm** [comment]
>Perhaps a better solution would be to write an empty file of the total size when creating the request instead of having continuously check whilst downloading?

yeah this makes more sense to me - we'd be able to fail before download 99% of the file and then running out of space - but I wonder if I'm missing something that would have made that a preferable solution.
